### ISSUE TITLE
Changing time format crash OpenFit

### ISSUE THREAD

**Issue by saper-2** [jareddlc/OpenFit#8]
Hi,

I just installed, OpenFit a,d I tried to set up my GF but I have two issues, the first one (more annoying) is 12h clock (I hate this format and I'm gonna bitch about it :smile: a lot ).
This is a LogCat from the moment tapping Time Format in OpenFit :

```
10-11 22:19:22.198: D/OpenFit:OpenFitFragment(27368): Sending Intent: bluetooth:time:false
10-11 22:19:22.230: D/AndroidRuntime(27368): Shutting down VM
10-11 22:19:22.231: E/AndroidRuntime(27368): FATAL EXCEPTION: main
10-11 22:19:22.231: E/AndroidRuntime(27368): Process: com.solderbyte.openfit, PID: 27368
10-11 22:19:22.231: E/AndroidRuntime(27368): java.lang.RuntimeException: Error receiving broadcast Intent { act=bluetooth flg=0x10 (has extras) } in com.solderbyte.openfit.OpenFitService$3@7c8c082
10-11 22:19:22.231: E/AndroidRuntime(27368):    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:891)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at android.os.Handler.handleCallback(Handler.java:739)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at android.os.Handler.dispatchMessage(Handler.java:95)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at android.os.Looper.loop(Looper.java:148)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at android.app.ActivityThread.main(ActivityThread.java:5417)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at java.lang.reflect.Method.invoke(Native Method)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
10-11 22:19:22.231: E/AndroidRuntime(27368): Caused by: java.lang.ClassCastException: long[] cannot be cast to int[]
10-11 22:19:22.231: E/AndroidRuntime(27368):    at com.solderbyte.openfit.util.OpenFitTimeZoneUtil.getTransitions(OpenFitTimeZoneUtil.java:17)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at com.solderbyte.openfit.util.OpenFitTimeZoneUtil.prevTransition(OpenFitTimeZoneUtil.java:59)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at com.solderbyte.openfit.OpenFitApi.getCurrentTimeInfo(OpenFitApi.java:273)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at com.solderbyte.openfit.OpenFitService.sendTime(OpenFitService.java:475)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at com.solderbyte.openfit.OpenFitService.handleUIMessage(OpenFitService.java:252)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at com.solderbyte.openfit.OpenFitService$3.onReceive(OpenFitService.java:748)
10-11 22:19:22.231: E/AndroidRuntime(27368):    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:881)
10-11 22:19:22.231: E/AndroidRuntime(27368):    ... 7 more
10-11 22:19:22.233: W/ActivityManager(849):   Force finishing activity com.solderbyte.openfit/.OpenFitActivity
10-11 22:19:22.296: I/OpenGLRenderer(849): Initialized EGL, version 1.4
10-11 22:19:22.371: I/art(849): Background partial concurrent mark sweep GC freed 22379(1281KB) AllocSpace objects, 16(320KB) LOS objects, 33% free, 31MB/47MB, paused 1.458ms total 122.313ms
10-11 22:19:22.757: W/ActivityManager(849): Activity pause timeout for ActivityRecord{46a6b65 u0 com.solderbyte.openfit/.OpenFitActivity t88 f}
10-11 22:19:23.357: W/OpenGLRenderer(22616): Incorrectly called buildLayer on View: ShortcutAndWidgetContainer, destroying layer...
10-11 22:19:23.357: W/OpenGLRenderer(22616): Incorrectly called buildLayer on View: ShortcutAndWidgetContainer, destroying layer...
10-11 22:19:33.117: W/ActivityManager(849): Activity destroy timeout for ActivityRecord{46a6b65 u0 com.solderbyte.openfit/.OpenFitActivity t88 f}
10-11 22:19:45.387: I/Process(27368): Sending signal. PID: 27368 SIG: 9
10-11 22:19:45.396: E/Surface(849): getSlotFromBufferLocked: unknown buffer: 0x93aaf220
10-11 22:19:45.403: I/Keyboard.Facilitator(1329): onFinishInput()
10-11 22:19:45.415: D/bt_btm_pm(17471): btm_pm_snd_md_req switching from SNIFF to ACTIVE.
10-11 22:19:45.415: E/bt_btif_sock_rfcomm(17471): find_rfc_slot_by_id unable to find RFCOMM slot id: 6
10-11 22:19:45.415: E/bt_rfcomm(17471): p_data_co_callback DATA_CO_CALLBACK_TYPE_INCOMING_SIZE failed, available:0
10-11 22:19:45.415: E/bt_btif_sock_rfcomm(17471): find_rfc_slot_by_id unable to find RFCOMM slot id: 6
10-11 22:19:45.415: D/GraphicsStats(849): Buffer count: 7
10-11 22:19:45.487: I/ActivityManager(849): Process com.solderbyte.openfit (pid 27368) has died
10-11 22:19:45.487: W/ActivityManager(849): Service crashed 2 times, stopping: ServiceRecord{7d4ae62 u0 com.solderbyte.openfit/.OpenFitService}
10-11 22:19:45.487: W/ActivityManager(849): Service crashed 2 times, stopping: ServiceRecord{3360504 u0 com.solderbyte.openfit/.NotificationService}
10-11 22:19:45.560: D/bt_btm_pm(17471): btm_pm_proc_mode_change switched from UNKNOWN to ACTIVE.
10-11 22:19:45.591: W/AppOps(849): Finishing op nesting under-run: uid 1000 pkg android code 24 time=0 duration=0 nesting=0
10-11 22:19:45.593: W/bt_btif(17471): bta_jv_rfc_port_to_cb(port_handle:0x9):jv handle:0x0 not FOUND
10-11 22:19:45.593: W/bt_rfcomm(17471): port_rfc_closed RFCOMM connection in state 3 closed: Closed (res: 19)
10-11 22:19:48.788: I/bt_btm_sec(17471): btm_sec_disconnected clearing pending flag handle:12 reason:19
10-11 22:19:48.790: E/BluetoothRemoteDevices(17471): state12newState1
10-11 22:19:48.801: D/BluetoothMapService(17471): onReceive
10-11 22:19:48.801: D/BluetoothMapService(17471): onReceive: android.bluetooth.device.action.ACL_DISCONNECTED
10-11 22:19:48.822: I/Coffee-BluetoothConnectionTracker(1762): Bluetooth disconnection broadcast for Gear Fit (8F56)
10-11 22:19:48.826: I/BTConnectionReceiver(21353): onReceive(context, Intent { act=android.bluetooth.device.action.ACL_DISCONNECTED flg=0x4000010 cmp=com.google.android.googlequicksearchbox/com.google.android.search.core.service.BluetoothConnectionReceiver (has extras) }, [BluetoothDevice: address=18:22:7E:D5:8F:56, alias=null, name=Gear Fit (8F56), majorDeviceClass=1792, deviceClass=1796]
10-11 22:19:48.827: I/BluetoothClassifier(21353): Bluetooth Device Name: Gear Fit (8F56)
```

I'm running OpenFit on: Nexus 5 with Android 6.0.0 (build MRA58K) .

**Comment by jareddlc** [comment]
I'll try to fix this asap

**Comment by jareddlc** [comment]
fixed in v1.4.5
